api-put-object-copy.go

```diff
@@ -0,0 +1,76 @@
+/*
+ * Minio Go Library for Amazon S3 Compatible Cloud Storage (C) 2016 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package minio
+
+import (
+	"net/http"
+	"strings"
+)
+
+// supportCopyObjectHeaders - list of supported headers
+var supportedCopyObjectHeaders = http.Header{
```

Comment:
> This can simply be a const string array. I would even move it inside CopyObject method if it is not useful outside in the future.

A string array requires for us to loop for each keys, to validate if its the right one. It is better to stick with map.